src/controller/python/test/test_scripts/base.py

```diff
@@ -133,13 +137,54 @@ def TestDiscovery(self, discriminator: int):
 
     def TestKeyExchange(self, ip: str, setuppin: int, nodeid: int):
         self.logger.info("Conducting key exchange with device {}".format(ip))
-        if not self.devCtrl.ConnectIP(ip.encode("utf-8"), setuppin, nodeid):
+        if not self.devCtrl.CommissionIP(ip.encode("utf-8"), setuppin, nodeid):
             self.logger.info(
                 "Failed to finish key exchange with device {}".format(ip))
             return False
         self.logger.info("Device finished key exchange.")
         return True
 
+    async def TestMultiFabric(self, ip: str, setuppin: int, nodeid: int):
+        self.logger.info("Opening Commissioning Window")
+
+        await self.devCtrl.SendCommand(nodeid, 0, Clusters.AdministratorCommissioning.Commands.OpenBasicCommissioningWindow(100))
+
+        self.logger.info("Creating 2nd Fabric Admin")
+        fabricAdmin2 = chip.FabricAdmin.FabricAdmin(fabricId=2, fabricIndex=2)
+
+        self.logger.info("Creating Device Controller on 2nd Fabric")
+        devCtrl2 = fabricAdmin2.NewController(self.controllerNodeId)
+
+        if not devCtrl2.CommissionIP(ip.encode("utf-8"), setuppin, nodeid):
+            self.logger.info(
+                "Failed to finish key exchange with device {}".format(ip))
+            return False
+
+        #
+        # Shut-down all the controllers (which will free them up as well as de-initialize the
+        # stack as well.
+        #
+        self.logger.info(
+            "Shutting down controllers & fabrics and re-initing stack...")
+
+        ChipDeviceCtrl.ChipDeviceController.ShutdownAll()
+        chip.FabricAdmin.FabricAdmin.ShutdownAll()
+
+        self.fabricAdmin = chip.FabricAdmin.FabricAdmin(
+            fabricId=1, fabricIndex=1)
+        fabricAdmin2 = chip.FabricAdmin.FabricAdmin(fabricId=2, fabricIndex=2)
+
+        self.devCtrl = self.fabricAdmin.NewController(self.controllerNodeId)
+        devCtrl2 = fabricAdmin2.NewController(self.controllerNodeId)
+
+        await self.devCtrl.ReadAttribute(nodeid, [(Clusters.Basic)])
```

Comment:
Can we read some fabric scoped attributes and verify the results? For example, try to read `Clusters.OperationalCredentials.Attributes.CurrentFabricIndex` instead of `Clusters.Basic` here?